ts/sagas/startup/watchApplicationActivitySaga.ts

```diff
@@ -84,9 +96,26 @@ export function* watchApplicationActivitySaga(): IterableIterator<ReduxSagaEffec
             timestamp: Date.now()
           };
 
+          const isAutomaticSessionRefreshEnabled = yield* select(
+            isAutomaticSessionRefreshEnabledSelector
+          );
+
           if (timeSinceLastStateChange >= backgroundActivityTimeoutMillis) {
             // The app was in background for a long time, request identification
             yield* put(identificationRequest());
+            // if the session refresh feature is active,
+            if (isAutomaticSessionRefreshEnabled) {
+              // after successful identification,
+              yield* take(identificationSuccess);
+              // the regeneration of the session token will be performed.
+              yield* put(
+                refreshSessionToken.request({
+                  withUserInteraction: false,
+                  showIdentificationModalAtStartup: false,
+                  showLoader: true
+                })
+              );
+            }
```

Comment:
We can extract this logic in its own generator function. What do you think?

Reply:
Done! Thanks for the suggestion!